perun-oidc-server/src/main/java/cz/muni/ics/oauth2/web/OAuthConfirmationController.java

```diff
@@ -114,36 +155,82 @@ public String confirmAccess(Map<String, Object> model, Principal p) {
 			return HttpCodeView.VIEWNAME;
 		}
 
-		if (prompts.contains("none")) {
+		if (prompts.contains(NONE)) {
 			// if we've got a redirect URI then we'll send it
+			return sendRedirect(authRequest, model, client);
+		}
 
-			String url = redirectResolver.resolveRedirect(authRequest.getRedirectUri(), client);
+		model.put(AUTH_REQUEST, authRequest);
+		model.put(CLIENT, client);
 
-			try {
-				URIBuilder uriBuilder = new URIBuilder(url);
+		String redirect_uri = authRequest.getRedirectUri();
+		model.put(REDIRECT_URI, redirect_uri);
```

Comment:
Inline the redirect_uri variable. Also, `redirect_uri` is not in the recommended naming format :).